snapcraft/cli/containers.py

```diff
@@ -32,25 +32,17 @@ def containerscli():
     "--debug", is_flag=True, help="Shells into the environment if the build fails."
 )
 def refresh(debug, **kwargs):
-    """Refresh an existing LXD container.
+    """Refresh existing packages.
 
     \b
     Examples:
         snapcraft refresh
-
-    This will take care of updating the apt package cache, upgrading packages
-    as needed as well as refreshing snaps.
     """
 
     build_environment = env.BuilderEnvironmentConfig()
-    if build_environment.is_host:
-        raise errors.SnapcraftEnvironmentError(
-            "The 'refresh' command only applies to LXD containers but "
-            "SNAPCRAFT_BUILD_ENVIRONMENT is not set or set to host.\n"
-            "Maybe you meant to update the parts cache instead? "
-            "You can do that with the following command:\n\n"
-            "snapcraft update"
-        )
-
-    project = get_project(**kwargs, debug=debug)
-    lxd.Project(project=project, output=None, source=os.path.curdir).refresh()
+
+    if build_environment.is_lxd:
+        project = get_project(**kwargs, debug=debug)
+        lxd.Project(project=project, output=None, source=os.path.curdir).refresh()
+    else:
+        repo.Repo.refresh_build_packages()
```

Comment:
It looks like this will run `sudo apt update` on the host instead of warning that refresh only applies to containers, is that expected?

Reply:
yes, refresh is a hidden command
we basically assume that running snapcraft at all might do sudo apt install too (these days)
We can polish this for 3.0 and remove all sudo calls as snapcraft will rerun itself with sudo (we cannot do that today as we will break)

Reply:
Works for me, just want to ensure it was on purpose.